app/kuma-dp/pkg/dataplane/accesslogs/server.go

```diff
@@ -0,0 +1,121 @@
+package accesslogs
+
+import (
+	"fmt"
+	"github.com/Kong/kuma/pkg/config/app/kuma-dp"
+	"github.com/Kong/kuma/pkg/core"
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/envoyproxy/go-control-plane/envoy/data/accesslog/v2"
+	v2 "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v2"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var logger = core.Log.WithName("accesslogs")
+
+type accessLogServer struct {
+	server *grpc.Server
+	conn   net.Conn
+}
+
+func NewAccessLogServer() *accessLogServer {
+	return &accessLogServer{
+		server: grpc.NewServer(),
+	}
+}
+
+func (a *accessLogServer) StreamAccessLogs(server v2.AccessLogService_StreamAccessLogsServer) error {
+	msg, err := server.Recv()
+	if err != nil {
+		return err
+	}
+
+	for _, httpLogEntry := range msg.LogEntries.(*v2.StreamAccessLogsMessage_HttpLogs).HttpLogs.LogEntry {
+		split := strings.Split(msg.GetIdentifier().GetLogName(), ";")
+		address := split[0]
+		format := split[1]
```

Comment:
`msg.GetIdentifier().GetLogName()` will only be present on the first message of gRPC stream => it must be cached and re-used for the remaining messages on that gRPC stream

Reply:
fixed